4/README.md

```diff
@@ -1,20 +1,41 @@
-Docker image for Varnish Cache 4.0 HTTP reverse proxy
+Varnish Cache 4.0 HTTP reverse proxy Docker image
 =====================================================
 
-This repository contains the source for Varnish Cache 4.0 Docker image.
+This container image includes Varnish 4.0 server and a reverse proxy for OpenShift and general usage.
+Users can choose between RHEL and CentOS based images.
+The RHEL image is available in the [Red Hat Container Catalog](https://access.redhat.com/containers/#/registry.access.redhat.com/rhscl/varnish-4-rhel7)
+as registry.access.redhat.com/rhscl/varnish-4-rhel7.
+The CentOS image is then available on [Docker Hub](https://hub.docker.com/r/centos/varnish-4-centos7/)
+as centos/varnish-4-centos7.
+
+
+Description
+-----------
+
+Varnish available as docker container is a base platform for 
+running Varnish server or building Varnish-based application. 
+Varnish Cache stores web pages in memory so web servers don't have to create 
+the same web page over and over again. Varnish Cache serves pages much faster 
+than any application server, giving the website a significant speed up.
+
 The image can be used as a base image for other applications based on Varnish Cache 4.0 or using s2i tool.
 
 
+Usage
+-----
+
+To build a simple [sample-app](https://github.com/sclorg/varnish-container/tree/master/4/test/test-app) application
+using standalone [S2I](https://github.com/openshift/source-to-image) and then run the
+resulting image with [Docker](http://docker.io) execute:
+
 *  **For RHEL based image**
     ```
-    $ docker pull registry.access.redhat.com/rhscl/varnish-4-rhel7
```

Comment:
I thought it might be unnecessary to pull it implicitly because s2i does it automatically. If there's no problem with just calling 
```
s2i build https://github.com/sclorg/varnish-container.git --context-dir=4/test/test-app/ rhscl/varnish-4-rhel7 sample-server
```
without `registry.access.redhat.com/` prefix which is also not used in other images like [here](https://github.com/ficap/mongodb-container/tree/new-readme-format/3.2). What do you think?